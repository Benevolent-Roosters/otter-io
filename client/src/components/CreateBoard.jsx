import React from 'react';
import { setCurrentBoard, setBoards } from '../redux/actionCreators.js';
import { connect } from 'react-redux';

const CreateBoard = (props) => {
  return (
    <div>This is CreateBoard</div>
  );
};

const mapStateToProps = (state) => {
  return {
    'boards': state.boards,
    'currentBoard': state.currentBoard
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    handleSetCurrentBoard(boardClicked) {
      dispatch(setCurrentBoard(boardClicked));
    },
    handleSetBoards(boardCreated) {
      dispatch(setBoards(boardCreated));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateBoard);

