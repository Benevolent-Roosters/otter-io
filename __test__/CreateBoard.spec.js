import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import CreateBoard, { UnwrappedCreateBoard }from '../client/src/components/CreateBoard.jsx';
import { Provider } from 'react-redux';
import rootReducer from '../client/src/redux/reducers.js';

describe('<CreateBoard />', () => {
  let store = {
    boards: [],
    currentBoard: {
      boardId: 0,
      teamName: '',
      repoName: '',
      owner: '',
      repoUrl: ''
    }
  };
  it ('should render the unconnected component correctly', () => {
    const wrapper = shallow(<UnwrappedCreateBoard />);
    expect(wrapper).toMatchSnapshot();
  });
  it ('should render the connected component correctly', () => {
    const wrapper = shallow(<Provider store={store}><CreateBoard /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
})