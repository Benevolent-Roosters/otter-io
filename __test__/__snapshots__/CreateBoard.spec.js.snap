// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateBoard /> should render the connected component correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(CreateBoard) />,
  "nodes": Array [
    <Connect(CreateBoard) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "boards": Array [],
                        "currentBoard": Object {
                          "boardId": 0,
                          "owner": "",
                          "repoName": "",
                          "repoUrl": "",
                          "teamName": "",
                        },
                      }
        }
>
        <Connect(CreateBoard) />
</Provider>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(CreateBoard) />,
          "store": Object {
            "boards": Array [],
            "currentBoard": Object {
              "boardId": 0,
              "owner": "",
              "repoName": "",
              "repoUrl": "",
              "teamName": "",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "boards": Array [],
          "currentBoard": Object {
            "boardId": 0,
            "owner": "",
            "repoName": "",
            "repoUrl": "",
            "teamName": "",
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(CreateBoard) />,
        "_debugID": 4,
        "_renderedOutput": <Connect(CreateBoard) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "boards": Array [],
            "currentBoard": Object {
              "boardId": 0,
              "owner": "",
              "repoName": "",
              "repoUrl": "",
              "teamName": "",
            },
          }
    }
>
    <Connect(CreateBoard) />
</Provider>,
}
`;

exports[`<CreateBoard /> should render the unconnected component correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    This is CreateBoard
</div>,
  "nodes": Array [
    <div>
      This is CreateBoard
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CreateBoard />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          This is CreateBoard
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          This is CreateBoard
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CreateBoard />,
}
`;
